.github/workflows/build.yaml

```diff
@@ -258,13 +286,77 @@ jobs:
         if: matrix.config.apt_get != ''
         run: sudo apt-get update && sudo apt-get install ${{ matrix.config.apt_get }}
 
+      - name: Prepare Windows Environment
+        if: runner.os == 'Windows'
+        shell: powershell
+        run: |
+          $ErrorActionPreference = 'Stop'
+
+          # Determine VS2019 install path
+          $vsInstallPath = (Get-VSSetupInstance | `
+            Select-VSSetupInstance -Product '*' -Latest -Version '[16.0,17.0)' `
+          ).InstallationPath
+
+          $vcvarsall = "${vsInstallPath}\VC\Auxiliary\Build\vcvarsall.bat"
+          if (-not (Test-Path -Path $vcvarsall)) {
+            Out-Error "Could not find Visual Studio setup script"
+            exit 1
+          }
+
+          # Run VS environment script
+          $env:VSCMD_SKIP_SENDTELEMETRY = '1'
+          cmd /c "`"${vcvarsall}`" ${{ matrix.config.msvc_arch }} & set" | foreach {
+            if ($_ -match '^\w+=') {
+              $k, $v = $_.split('=', 2)
+              Set-Item -Force -Path "env:\${k}" -Value $v
+            }
+          }
+
+          # The test suite currently requires that the compiler specified by the "CC"
+          # env variable is on a path without spaces. Provide that by creating a
+          # junction from ~/opt/llvm to the Visual Studio path.
```

Comment:
Sounds like that should be fixed instead?

Reply:
I think that'll be tricky unfortunately -- there seem to be loads of implicit assumptions in the `test/suites/*.bash` tests that all tools are in paths without spaces. Ccache itself seems to cope okay, with the exception of environment variables like CCACHE_COMPILERCHECK where you can't reference tools with spaces